tests/commands/test_shell.py

```diff
@@ -67,6 +71,26 @@ def test_write_bash_functions_file_exists(default_kwargs, tmp_path, capsys):
     assert "The integration is already present and enabled" in captured.out
 
 
+def test_write_bash_functions_missing_bashrc_d(default_kwargs, tmp_path, capsys):
+    original_home = os.environ["HOME"]
+    try:
+        os.environ["HOME"] = (tmp_path / "test_home").as_posix()
+        some_bash_file = tmp_path / "test.bashrc"
+        default_kwargs["text_renderer"] = create_text_renderer()
+        default_kwargs["warning_renderer"] = create_warning_renderer()
+        bash_shell_operation = BaseShellOperation(**default_kwargs)
+
+        bash_shell_operation._write_bash_functions(some_bash_file, "export TEST=1")
+
+        assert some_bash_file.exists()
+        assert some_bash_file.read_text() == "export TEST=1"
+        captured = capsys.readouterr()
+        assert "In order to use shell integration" in captured.err
+        assert oct(some_bash_file.stat().st_mode).endswith("600")
+    finally:
+        os.environ["HOME"] = original_home
```

Comment:
We can use `monkeypatch.setenv()` for this.

Reply:
Ah nice! I am not too familiar with monkey patching yet. I'll refactor.